test/unit-test/ota_utest.c

```diff
@@ -657,82 +657,85 @@ void test_OTA_ResumeWhenStopped()
 
     /* OTA agent should remain in stopped state. */
     otaWaitForEmptyEvent();
-    TEST_ASSERT_EQUAL( OtaAgentStateStopped, OTA_GetAgentState() );
+    TEST_ASSERT_EQUAL( OtaAgentStateStopped, OTA_GetState() );
 }
 
 void test_OTA_ResumeWhenSuspended()
 {
     otaGoToState( OtaAgentStateSuspended );
-    TEST_ASSERT_EQUAL( OtaAgentStateSuspended, OTA_GetAgentState() );
+    TEST_ASSERT_EQUAL( OtaAgentStateSuspended, OTA_GetState() );
 
     TEST_ASSERT_EQUAL( OTA_ERR_NONE, OTA_Resume() );
     otaWaitForState( OtaAgentStateRequestingJob );
-    TEST_ASSERT_EQUAL( OtaAgentStateRequestingJob, OTA_GetAgentState() );
+    TEST_ASSERT_EQUAL( OtaAgentStateRequestingJob, OTA_GetState() );
 }
 
 void test_OTA_ResumeWhenReady()
 {
     otaGoToState( OtaAgentStateReady );
-    TEST_ASSERT_EQUAL( OtaAgentStateReady, OTA_GetAgentState() );
+    TEST_ASSERT_EQUAL( OtaAgentStateReady, OTA_GetState() );
 
     /* Calling resume when OTA agent is not suspend state. This should be an unexpected event and
      * the agent should remain in ready state. */
     TEST_ASSERT_EQUAL( OTA_ERR_NONE, OTA_Resume() );
     otaWaitForEmptyEvent();
-    TEST_ASSERT_EQUAL( OtaAgentStateReady, OTA_GetAgentState() );
+    TEST_ASSERT_EQUAL( OtaAgentStateReady, OTA_GetState() );
 }
 
 void test_OTA_ResumeFailedWhenSuspended()
 {
     otaGoToState( OtaAgentStateSuspended );
-    TEST_ASSERT_EQUAL( OtaAgentStateSuspended, OTA_GetAgentState() );
+    TEST_ASSERT_EQUAL( OtaAgentStateSuspended, OTA_GetState() );
 
     /* Set the event send interface to a mock function that always fail. */
     otaInterfaces.os.event.send = mockOSEventSendAlwaysFail;
 
     /* Resume should fail and OTA agent should remain in suspend state. */
     TEST_ASSERT_EQUAL( OTA_ERR_EVENT_Q_SEND_FAILED, OTA_Resume() );
-    TEST_ASSERT_EQUAL( OtaAgentStateSuspended, OTA_GetAgentState() );
+    TEST_ASSERT_EQUAL( OtaAgentStateSuspended, OTA_GetState() );
 }
 
 void test_OTA_Statistics()
 {
     otaGoToState( OtaAgentStateReady );
-    TEST_ASSERT_EQUAL( OtaAgentStateReady, OTA_GetAgentState() );
+    TEST_ASSERT_EQUAL( OtaAgentStateReady, OTA_GetState() );
 
-    TEST_ASSERT_EQUAL( 0, OTA_GetPacketsDropped() );
-    TEST_ASSERT_EQUAL( 0, OTA_GetPacketsQueued() );
-    TEST_ASSERT_EQUAL( 0, OTA_GetPacketsProcessed() );
-    TEST_ASSERT_EQUAL( 0, OTA_GetPacketsReceived() );
+    OtaAgentStatistics_t statistics = { 0 };
+    TEST_ASSERT_EQUAL( OTA_ERR_NONE, OTA_GetStatistics( &statistics ) );
```

Comment:
We need to test both success and failure case (see comment above) here.  Also, lines 704-709 aren't verifying anything about what `OTA_GetStatistics()` has done since `statistics` was initialized to 0.  `statistics` should be initialized to something other than the values returned by `OTA_GetStatistics()`, and this test should initialize each member in the returned `OtaAgentStatistics_t` object such that lines 704-709 can verify that each was assigned correctly.

Reply:
Added check for failure: https://github.com/aws/ota-for-aws-iot-embedded-sdk/blob/9b8ba39796aab5af8d52b3423c4a497ae0fbbe4a/test/unit-test/ota_utest.c#L703

We would add further tests to check the correctness of the API with the OTA Tests PR.

Reply:
This will be fixed in a separate PR